### PR TITLE
Readme, API

### LINKED ISSUE #56: Create README file

### PR DESCRIPTION
**Pull request by talyguryn**
close #56

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,231 @@
...
+
+    * [Example](#example)
+
+* [Get image](#get-image)
+
+    * [Filters](#filters)
+
+## Usage
+
+1. Open Capella or use API to upload image.
```
Comment (PolinaShneider) [inline README.md]: Сюда б ссылку

File: README.md [hunk 2]
```diff
@@ -1 +1,231 @@
...
+
+2. Get image by given url with applied filters.
+
+### File requirements
+
+Max size for the file is `15MB`.
+
+At this time we support these types of images:
```
Comment (PolinaShneider) [inline README.md]: Мб, "At this time" -> "Now"

File: README.md [hunk 3]
```diff
@@ -1 +1,231 @@
...
+#### List of messages for failed requests
+
+| Message                          |  Description                          |
+|----------------------------------|---------------------------------------|
+| `Method not allowed`             | Request method is not POST            |
+| `File or link is missing`        | No expected data was found            |
+| `File is missing`                | File name is empty                    |
+| `Link is missing`                | Field link is empty                   |
+| `Wrong source mime-type`         | No support file with this mime-type   |
+| `Source is too big`              | File size is more than a limit        |
```
Comment (PolinaShneider) [inline README.md]: File size exceeds the limit

File: README.md [hunk 4]
```diff
@@ -1 +1,231 @@
...
+
+Scale the image to the largest size such that both its width and its height can fit inside the target rectangle.
+
+| Param    | Type    | Description                                                  |
+|----------|---------|--------------------------------------------------------------|
+| `width`  | Integer | Max image width or max image box size if no height was given |
+| `height` | Integer | (optional) Max image height                                  |
+
+`https://capella.ifmo.su/<image_id>/resize/300x400`
+`https://capella.ifmo.su/<image_id>/resize/500`
```
Comment (PolinaShneider) [inline README.md]: Сделай так, чтобы был перенос на новую строку

File: README.md [hunk 5]
```diff
@@ -1 +1,231 @@
...
+| `height` | Integer | (optional) Target rectangle height                                         |
+
+`https://capella.ifmo.su/<image_id>/crop/300x400`
+`https://capella.ifmo.su/<image_id>/crop/250`
+
+##### Additional params
+
+If you need to crop area from specified point then pass these params.
+
+Note that this way `width` and `height` will be params for cropped area size.
```
Comment (PolinaShneider) [inline README.md]: Note that this way width and height will be size params for the cropped area

**Review by PolinaShneider — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,233 @@
...
+
+    * [Example](#example)
+
+* [Get image](#get-image)
+
+    * [Filters](#filters)
+
+## Usage
+
+1. Open [capella.ifmo.su](https://capella.ifmo.su) or use [API](#upload-api) to upload image.
```
Comment (neSpecc) [inline README.md]: to upload an image.

File: README.md [hunk 7]
```diff
@@ -1 +1,233 @@
-# capella
\ No newline at end of file
+# [Сapella](https://capella.ifmo.su)
+
+Cloud service for image storage and delivery. Upload files and accept image-filters on the fly with simple API.
```
Comment (neSpecc) [inline README.md]: with the simple API

File: README.md [hunk 8]
```diff
@@ -1 +1,233 @@
...
+
+* [Get image](#get-image)
+
+    * [Filters](#filters)
+
+## Usage
+
+1. Open [capella.ifmo.su](https://capella.ifmo.su) or use [API](#upload-api) to upload image.
+
+2. Get image by given url with applied filters.
```
Comment (neSpecc) [inline README.md]: URL

File: README.md [hunk 9]
```diff
@@ -1 +1,233 @@
...
+
+### File requirements
+
+Max size for the file is `15MB`.
```
Comment (neSpecc) [inline README.md]: Maximum

File: README.md [hunk 10]
```diff
@@ -1 +1,233 @@
...
+- gif
+- bmp
+
+Please note that each uploaded file would be converted to PNG.
+
+## Upload API
+
+### Request
+
+Per one request you can upload one file or link.
```
Comment (neSpecc) [inline README.md]: You can upload one file or URL per one request.

File: README.md [hunk 11]
```diff
@@ -1 +1,233 @@
...
+
+You can upload image file or send link to image from your app by making a request to `https://capella.ifmo.su/upload`.
```
Comment (neSpecc) [inline README.md]: link to the  image

File: README.md [hunk 12]
```diff
@@ -1 +1,233 @@
...
+
+| Method | URI      | Data                      |
+|--------|----------|---------------------------|
+| `POST` | `upload` | file in `file` field      |
+| `POST` | `upload` | image url in `link` field |
+
+You will get a json response from server.
```
Comment (neSpecc) [inline README.md]: JSON

File: README.md [hunk 13]
```diff
@@ -1 +1,233 @@
...
+| `message` | String  | Result message    |
+
+#### Success
+
+| Field     | Type    | Description or value                   |
+|-----------|---------|----------------------------------------|
+| `success` | Boolean | `true`                                 |
+| `message` | String  | `Image uploaded`                       |
+| `id`      | String  | Image id                               |
+| `url`     | String  | Full link to uploaded image on Capella |
```
Comment (neSpecc) [inline README.md]: Full link to the uploaded image

File: README.md [hunk 14]
```diff
@@ -1 +1,233 @@
...
+
+r = requests.post('https://capella.ifmo.su/upload', data=data)
+response = json.loads(r.content)
+
+print(response)
+```
+
+## Get image
+
+You can get each uploaded image by the following url scheme.
```
Comment (neSpecc) [inline README.md]: URL

File: README.md [hunk 15]
```diff
@@ -1 +1,233 @@
...
+
+`https://capella.ifmo.su/<image_id>`
+
+### Filters
+
+Add filters at the end of request image URL.
```
Comment (neSpecc) [inline README.md]: You can apply filters by adding them at the end of the image URL.

File: README.md [hunk 16]
```diff
@@ -1 +1,233 @@
...
+
+#### Resize
+
+Scale the image to the largest size such that both its width and its height can fit inside the target rectangle.
+
+| Param    | Type    | Description                                                  |
+|----------|---------|--------------------------------------------------------------|
+| `width`  | Integer | Max image width or max image box size if no height was given |
```
Comment (neSpecc) [inline README.md]: Max -> Maximum
max -> maximum
box -> square

File: README.md [hunk 17]
```diff
@@ -1 +1,233 @@
...
+| `height` | Integer | (optional) Max image height                                  |
```
Comment (neSpecc) [inline README.md]: Maximum

File: README.md [hunk 18]
```diff
@@ -1 +1,233 @@
...
+
+`https://capella.ifmo.su/<image_id>/resize/500`
+
+#### Crop
+
+Cover the target rectangle by the image. Nice tool for creating covers or profile pics.
+
+| Param    | Type    | Description                                                                |
+|----------|---------|----------------------------------------------------------------------------|
+| `width`  | Integer | Target rectangle width or target rectangle box size if no height was given |
```
Comment (neSpecc) [inline README.md]: rectangle's width
rectangle's box
box -> square

File: README.md [hunk 19]
```diff
@@ -1 +1,233 @@
...
+| `height` | Integer | (optional) Target rectangle height                                         |
+
+`https://capella.ifmo.su/<image_id>/crop/300x400`
+
+`https://capella.ifmo.su/<image_id>/crop/250`
+
+##### Additional params
+
+If you need to crop area from specified point then pass these params.
```
Comment (neSpecc) [inline README.md]: an area

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: README.md [hunk 20]
```diff
@@ -1 +1,233 @@
...
+
+1. Open [capella.ifmo.su](https://capella.ifmo.su) or use [API](#upload-api) to upload image.
+
+2. Get image by given url with applied filters.
+
+### File requirements
+
+Max size for the file is `15MB`.
+
+Capella supports these types of images:
```
Comment (khaydarov) [inline README.md]: Supports the following types

File: README.md [hunk 21]
```diff
@@ -1 +1,233 @@
...
+
+`https://capella.ifmo.su/<image_id>/crop/300x400`
+
+`https://capella.ifmo.su/<image_id>/crop/250`
+
+##### Additional params
+
+If you need to crop area from specified point then pass these params.
+
+Note that this way `width` and `height` will be size params for the cropped area.
```
Comment (khaydarov) [inline README.md]: Note that width and height are params that describes cropped area

**Review by khaydarov — CHANGES_REQUESTED** [review]

**Review by PolinaShneider — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Review by khaydarov — APPROVED** [review]

**Review by gohabereg — APPROVED** [review]

**Pull request merged**